pkg/apiserver/experiment/experiment.go

```diff
@@ -63,3 +68,192 @@ func (s *Service) getExperimentDetail(c *gin.Context) {}
+	return nil
+}
+
+// getTimeChaosState returns the state of TimeChaos
+func (s *Service) getTimeChaosState (stateInfo map[string]int) error {
+	var chaosList v1alpha1.TimeChaosList
+	err := s.kubeCli.List(context.Background(), &chaosList)
+	if err != nil {
+		return err
+	}
+	for _, chaos := range chaosList.Items {
+		stateInfo["Total"]++
+		switch chaos.Status.ChaosStatus.Experiment.Phase {
+		case v1alpha1.ExperimentPhaseRunning:
+			stateInfo["running"]++
+		case v1alpha1.ExperimentPhasePaused:
+			stateInfo["Paused"]++
+		case v1alpha1.ExperimentPhaseFailed:
+			stateInfo["Failed"]++
+		case v1alpha1.ExperimentPhaseFinished:
+			stateInfo["Finished"]++
+		}
+	}
+	return nil
+}
+
+// getKernelChaosState returns the state of KernelChaos
+func (s *Service) getKernelChaosState (stateInfo map[string]int) error {
+	var chaosList v1alpha1.KernelChaosList
+	err := s.kubeCli.List(context.Background(), &chaosList)
+	if err != nil {
+		return err
+	}
+	for _, chaos := range chaosList.Items {
+		stateInfo["Total"]++
+		switch chaos.Status.ChaosStatus.Experiment.Phase {
+		case v1alpha1.ExperimentPhaseRunning:
+			stateInfo["running"]++
+		case v1alpha1.ExperimentPhasePaused:
+			stateInfo["Paused"]++
+		case v1alpha1.ExperimentPhaseFailed:
+			stateInfo["Failed"]++
+		case v1alpha1.ExperimentPhaseFinished:
+			stateInfo["Finished"]++
+		}
+	}
+	return nil
+}
+
+// getStressChaosState returns the state of StressChaos
+func (s *Service) getStressChaosState (stateInfo map[string]int) error {
+	var chaosList v1alpha1.StressChaosList
+	err := s.kubeCli.List(context.Background(), &chaosList)
+	if err != nil {
+		return err
+	}
+	for _, chaos := range chaosList.Items {
+		stateInfo["Total"]++
+		switch chaos.Status.ChaosStatus.Experiment.Phase {
+		case v1alpha1.ExperimentPhaseRunning:
+			stateInfo["running"]++
+		case v1alpha1.ExperimentPhasePaused:
+			stateInfo["Paused"]++
+		case v1alpha1.ExperimentPhaseFailed:
+			stateInfo["Failed"]++
+		case v1alpha1.ExperimentPhaseFinished:
+			stateInfo["Finished"]++
+		}
+	}
+	return nil
+}
+
+func (s *Service) state (c *gin.Context) {
+	data := make(map[string]int)
+	data["Total"] = 0
+	data["Running"] = 0
+	data["Paused"] = 0
+	data["Failed"] = 0
+	data["Finished"] = 0
+	getChaosWrong := gin.H{
+		"status": 1003,
+		"message": "get chaos wrong",
+		"data": make(map[string]int),
+	}
+
+	err := s.getPodChaosState(data)
+	if err != nil {
+		c.JSON(http.StatusOK, getChaosWrong)
+		return
+	}
+	err = s.getIoChaosState(data)
+	if err != nil {
+		c.JSON(http.StatusOK, getChaosWrong)
+		return
+	}
+	err = s.getNetworkChaosState(data)
+	if err != nil {
+		c.JSON(http.StatusOK, getChaosWrong)
+		return
+	}
+	err = s.getTimeChaosState(data)
+	if err != nil {
+		c.JSON(http.StatusOK, getChaosWrong)
+		return
+	}
+	err = s.getKernelChaosState(data)
+	if err != nil {
+		c.JSON(http.StatusOK, getChaosWrong)
+		return
+	}
+	err = s.getStressChaosState(data)
+	if err != nil {
+		c.JSON(http.StatusOK, getChaosWrong)
+		return
+	}
```

Comment:
I think we can optimize these `getChaosState` functions. IMO what about adding a new interface for these different chaos lists? Then we can just use one function like `s.getChaosState(data, chaosList <newInterfaceName>)`. And in this way, we can also use errGroup to make the query parallel.

Reply:
But IMO this is okay to put in next PR.